src/sentry/models/monitorenvironment.py

```diff
@@ -30,3 +31,30 @@ class Meta:
         indexes = [models.Index(fields=["monitor", "environment"])]
 
     __repr__ = sane_repr("monitor_id", "environment_id")
+
+    def mark_failed(self, last_checkin=None, reason=MonitorFailure.UNKNOWN):
```

Comment:
duplicating this for now, sans the event creation logic

Reply:
Sounds good, we'll want to consolidate into a `logic` module or something at some point I think?

Reply:
yeah we have some design decisions around `Monitor` status as an aggregate of `MonitorEnvironment`, or we can just use the most recent check in